workflow/common/util.go

```diff
@@ -324,6 +324,23 @@ func IsDone(un *unstructured.Unstructured) bool {
 		un.GetLabels()[LabelKeyWorkflowArchivingStatus] != "Pending"
 }
 
+// Remove ASAP in 3.6, used **only** for backward compatability.
```

Comment:
It would probably make more sense to PR this to the `release-3.5` branch then? Or is part of this meant to stay on `main`?

Reply:
Depends on how long we want to support  this fallback fix for, was hoping for agreement that we do not expect workflows initially run with 3.4 to finish to completion when the controller is upgraded to 3.6 while workflows are running.

Reply:
Yea I can agree to that

Reply:
Done!